packages/core/core/src/services/document-service/index.ts

```diff
@@ -29,11 +31,15 @@ export const createDocumentService = (strapi: Strapi): any => {
       return repositories.get(uid)!;
     }
 
-    const repository = createContentTypeRepository(strapi.contentType(uid));
-    repositories.set(uid, repository);
+    const contentType = strapi.contentType(uid);
+    const repository = createContentTypeRepository(contentType);
+
+    repositories.set(uid, middlewares.wrapObject(repository, { contentType }));
```

Comment:
what's the benefit of having a repository per uid, rather than one for single types and another for collection types?

Reply:
👍 makes sense 😄